lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -993,7 +993,23 @@ def sanitize_network_params(self):
                                           " a VLAN name under VM network list.")
 
             if 'name' in network and find_obj(self.content, [vim.Network], network['name']) is None:
-                self.module.fail_json(msg="Network '%(name)s' does not exist." % network)
+                netfound = False
+                host = None
+                vm = self.get_vm()
+                if vm:
+                    host = vm.summary.runtime.host
+                elif self.params['esxi_hostname']:
+                    host = self.select_host()
+
```

Comment:
How about adding this -

```
diff --git a/lib/ansible/modules/cloud/vmware/vmware_guest.py b/lib/ansible/modules/cloud/vmware/vmware_guest.py
index 6985ca5497..f4fec7eec5 100644
--- a/lib/ansible/modules/cloud/vmware/vmware_guest.py
+++ b/lib/ansible/modules/cloud/vmware/vmware_guest.py
@@ -992,8 +992,24 @@ class PyVmomiHelper(PyVmomi):
                 self.module.fail_json(msg="Please specify at least a network name or"
                                           " a VLAN name under VM network list.")

-            if 'name' in network and find_obj(self.content, [vim.Network], network['name']) is None:
-                self.module.fail_json(msg="Network '%(name)s' does not exist." % network)
+            if 'name' in network:
+                if find_obj(self.content, [vim.Network], network['name']) is None:
+                    self.module.fail_json(msg="Network '%(name)s' does not exist." % network)
+                else:
+                    if self.params.get('esxi_hostname', None):
+                        esxi_hostname = self.select_host()
+                        if network['name'] not in [nw.name for nw in esxi_hostname.network]:
+                            self.module.fail_json(msg="Specified network '%s' is not associated "
+                                                      "with the given ESXi hostname '%s'" % (network['name'],
+                                                                                             self.params['esxi_hostname']))
+                    elif self.params.get('cluster', None):
+                        cluster = self.cache.get_cluster(self.params['cluster'])
+                        if network['name'] not in [nw.name for nw in cluster.network]:
+                            self.module.fail_json(msg="Specified network '%s' is not associated"
+                                                      " with the given cluster '%s'" % (network['name'],
+                                                                                        self.params['cluster']))
```

Reply:
```
+                if find_obj(self.content, [vim.Network], network['name']) is None:
+                    self.module.fail_json(msg="Network '%(name)s' does not exist." % network)
+                else:
```
^ The `else` will get evaluated when the network **is** found, which is not what we're after. If the network **is not** found, we want to try searching the host for it instead.

```
+                    if self.params.get('esxi_hostname', None):
+                        esxi_hostname = self.select_host()
```

This will only handle the scenario where the user specifies an esxi_hostname. It won't handle the scenario where the user didn't specify one, but we need to check what host it was automatically assigned to.

```
+                        cluster = self.cache.get_cluster(self.params['cluster'])
+                        if network['name'] not in [nw.name for nw in cluster.network]:
```
I don't follow this one. `get_cluster` looks like it will only ever return `vim.ClusterComputeResource` objects. This object does not have a `network` attribute.